docs/application/native/guides/device/diagnostics.md

```diff
@@ -122,44 +125,137 @@ void crash_callback(diagnostics_ctx_h ctx, void *user_data)
 
-Diagnostics API also can request data from other application or service, especially the diagnostic data that for various reasons cannot or must not be saved in the logs, such as binary or in custom format data, large volume data, reports that are time-consuming to create, and so on.
+Sometimes there is a need for getting data related to some specific diagnostics event. For example, Bugreport service sends an event that a new bug report has been created and the client wants to get the bug report's content after receiving the event. If the application supports event-specific data request, then `diagnostics_get_data()` must be used.
 
-To do so, call `diagnostics_request_client_data()`. The `client_id` variable must be the id of the process that supports Diagnostics API requests:
+# Current implementations
+
+## Bugreport service
+
+Bugreport service provides a diagnostic interface to allow you to read the bug reports in case the process in the system fails. To do so, you must register the callback `diagnostics_set_notification_cb`, it is called when the bug report is created:
 
 ```cpp
-void request_data()
+void notification_handler(diagnostics_ctx_h ctx, void *user_data)
+{
+    ....
+}
+
+int main(int argc, char **argv)
+{
+    diagnostics_set_notification_cb(notification_handler, NULL);
+    diagnostics_subscribe_event("BugreportCreated", "org.tizen.bugreport-service");
+}
+```
+
+To deregister the callback, use `diagnostics_unset_notification_cb()`.
+
+In case a new bug report is created, callback is called. To read the content of the bug report use `diagnostics_get_data()`:
+
+```cpp
+void notification_handler(diagnostics_ctx_h ctx, void *user_data)
 {
-    const char *client_id = "org.tizen.some_service";
     diagnostics_data_h data;
-    const char *params[] = {"custom_parameter"};
+    const char *params[] = {"--type", "crash-info-json"};
 
-    int ret = diagnostics_request_client_data(client_id, params, 1, &data);
-    if (ret != DIAGNOSTICS_ERROR_NONE)
+    diagnostics_get_data(ctx, params, 2, &data);
+
+    ...
+
+    diagnostics_destroy(ctx);
+}
+```
+
+Use the `param` argument to specify which data to read from the bug report. Currently, the proper values are:
+
+-   `--type <report_type>` - Specify the report type. Available types are:
+    -   `"bugreport"` - Bugreport file is returned, which contains information about system such as system logs, list of processes, and so on. The output can be modified by dump_systemstate configuration (default).
+    -   `"crash-info"` - Returned report contains information about crashed process such as callstack, mapped memory regions, and so on.
+    -   `"crash-info-json"` - crash-info report in JSON format.
+    -   `"fullreport"` - Full crash report ZIP archive is collected (if single report specified).
+    -   `"coredumptar"` - Coredump file in TAR archive (if single report specified).
+-   `--last <seconds>` - Get reports generated with the last <seconds>
```

Comment:
sorry I missed within

`Get reports generated within the last <seconds>.`

Reply:
@mijin85cho, as you can see in my comment at that time there was no period or seconds missing, it is been updated post my comment I suppose.